### PR TITLE
Added optgroups filter and tests

### PR DESCRIPTION
**Pull request by smithdc1**
Supersedes #1121 

This is slimmed down to just the new filter and tests. 

Needed to add `values = field.field.widget.format_value(field.value())` compared to the previous version to correctly format the values. You'd see the test where an integer is passed in would fail without this.

EDIT: 
I'm assuming this is easier to review without template changes. That can come as follow-up PRs. Do let me know if you disagree!

David

### CODE REVIEW

**Review by carltongibson — APPROVED** [review]
Hey @smithdc1 — this looks fine as an addition. 

Is it purely internal, or does it warrant docs?

**Comment by smithdc1** [comment]
It's internal in that regular users won't be using it, but I guess folk making their own template packs will find it useful.

I think a doc string could be the right answer here.

I've just realised that I don't often write doc strings. I need to think about this more.

File: crispy_forms/templatetags/crispy_forms_filters.py [hunk 1]
```diff
@@ -125,3 +125,34 @@ def as_crispy_field(field, template_pack=TEMPLATE_PACK, label_class="", field_cl
...
     return mark_safe(flatatt(attrs))
+
+
+@register.filter
+def optgroups(field):
+    """
+    A template filter to help rendering of fields with optgroups.
+
+    Returns:
+        A tuple of label, option, index
```
Comment (carltongibson) [inline crispy_forms/templatetags/crispy_forms_filters.py:136]: Does it not return a list of such tuples? 🤔

**Review by carltongibson — COMMENTED** [review]
Hi @smithdc1 — good docstring. 

This looks great.

**Comment by smithdc1** [comment]
Thanks Carlton!

**Pull request merged**